idunn/geocoder/nlu_client.py

```diff
@@ -135,17 +178,31 @@ def fuzzy_match(cls, query, response):
             logger.error("Request to NLU tokenizer failed", exc_info=True)
             return []
 
-        intentions = []
         tags_list = [t for t in response_nlu.json()["NLU"] if t["tag"] != "O"]
-        tags_count = Counter(t["tag"] for t in tags_list)
-        if tags_count == {"city": 1, "cat": 1}:
-            # 1 category + 1 place
-            intention = await self.build_intention_category_place(tags_list, lang=lang)
+
+        if self.is_poi_request(tags_list):
+            return []
+
+        intentions = []
+        brand_query = self.build_brand_query(tags_list)
+        cat_query = self.build_category_query(tags_list)
+        place_query = self.build_place_query(tags_list)
+        skip_classifier = brand_query is not None
+
+        if (bool(brand_query) ^ bool(cat_query)) and place_query:
```

Comment:
So if I understand correctly, you only enter in here if only one of brand_query or cat_query is true. I think it might be worth adding a comment to make the reading simpler.

Reply:
I just updated it with this comment:

```python
# If there is a label for both a category and a brand, we consider that the request is
# ambiguous and ignore both.
if bool(brand_query) ^ bool(cat_query):
    ...
```